acceptance/tests/facts/debian.rb

```diff
@@ -0,0 +1,119 @@
+test_name "Facts should resolve as expected in Debian 6 and 7"
+
+#
+# This test is intended to ensure that facts specific to an OS configuration
+# resolve as expected in Debian 6 and 7.
+#
+# Facts tested: os, processors, ruby, networking, identity, kernel
+#
+
+confine :to, :platform => /debian-squeeze|debian-wheezy/
+
+agents.each do |agent|
+  os_version = agent['platform'] =~ /squeeze/ ? '6' : '7'
+  if agent['platform'] =~ /x86_64/
+    os_arch     = 'amd64'
+    os_hardware = 'x86_64'
+  else
+    os_arch     = 'i386'
+    os_hardware = 'i686'
+  end
+
+  step "Ensure the OS fact resolves as expected"
+  expected_os = {
+                  'os.architecture'         => os_arch,
+                  'os.distro.codename'      => os_version == '6' ? 'squeeze' : 'wheezy',
+                  'os.distro.description'   => /Debian GNU\/Linux #{os_version}\.\d/,
+                  'os.distro.id'            => 'Debian',
+                  'os.distro.release.full'  => /#{os_version}\.\d/,
+                  'os.distro.release.major' => os_version,
+                  'os.distro.release.minor' => /\d/,
+                  'os.family'               => 'Debian',
+                  'os.hardware'             => os_hardware,
+                  'os.name'                 => 'Debian',
+                  'os.release.full'         => /#{os_version}\.\d/,
+                  'os.release.major'        => os_version,
+                  'os.release.minor'        => /\d/,
+                }
+
+  expected_os.each do |fact, value|
+    assert_match(value, fact_on(agent, fact))
+  end
+
+  step "Ensure the Processors fact resolves with reasonable values"
+  expected_processors = {
+                          'processors.count'         => /[1-9]/,
+                          'processors.physicalcount' => /[1-9]/,
+                          'processors.isa'           => /unknown|#{os_hardware}/,
+                          'processors.models'        => /"Intel\(R\).*"/
+                        }
+
+  expected_processors.each do |fact, value|
+    assert_match(value, fact_on(agent, fact))
+  end
+
+  if os_version == 7
```

Comment:
Some of the older platforms (like Debian 6) have an older Ruby by default, and so the Ruby fact doesn't resolve.